File: src/main/java/de/lulonaut/Bot/commands/config/ConfigCommand.java
package de.lulonaut.Bot.commands.config;

import de.lulonaut.Bot.utils.Conf;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;

import java.util.Objects;

public class ConfigCommand extends ListenerAdapter {

    @Override
    public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
        if (event.getAuthor().isBot() || !event.getMessage().getContentRaw().equalsIgnoreCase(Conf.PREFIX + "config")) {
            return;
        } else {
            if (!Objects.requireNonNull(event.getMember()).getPermissions().contains(Permission.MANAGE_SERVER)) {
                EmbedBuilder eb = new EmbedBuilder();
                eb.setTitle("Error!");
                eb.setDescription("To use this command you need to have the \"Manage Server\" permission!");
                event.getChannel().sendMessage(eb.build()).queue();
                return;
            }

            System.out.println("starting state machine");
            ConfigStateMachine csm = new ConfigStateMachine(event.getChannel(), event, event.getMember());
            event.getJDA().addEventListener(csm);
            new Thread(() -> {
                try {
                    //timeout of 5 minutes
                    System.out.println("going to sleep");
                    Thread.sleep(300000);
                    System.out.println("woke up");
                    System.out.println(csm.done);
                    if (!csm.done) {
                        System.out.println("not done, cancelling");
                        event.getChannel().sendMessage("Timeout reached, canceling config.").queue();
                        event.getJDA().removeEventListener(csm);
                    }
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }).start();
        }

    }
}

File: src/main/kotlin/de/lulonaut/bot/commands/config/ConfigCommand.kt
package de.lulonaut.bot.commands.config

import de.lulonaut.bot.cache.DatabaseCache
import de.lulonaut.bot.utils.Conf
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter
import java.util.*

class ConfigCommand : ListenerAdapter() {
    override fun onGuildMessageReceived(event: GuildMessageReceivedEvent) {
        var prefix = DatabaseCache.getConfig(event.guild.id)?.get("prefix")
        if (prefix == null) {
            prefix = Conf.PREFIX
        }
        if (event.author.isBot || !event.message.contentRaw.equals(prefix + "config", ignoreCase = true)) {
            return
        } else {
            if (!Objects.requireNonNull(event.member)!!.permissions.contains(Permission.MANAGE_SERVER)) {
                val eb = EmbedBuilder()
                eb.setTitle("Error!")
                eb.setDescription("To use this command you need to have the \"Manage Server\" permission!")
                event.channel.sendMessage(eb.build()).queue()
                return
            }
            val csm = ConfigStateMachine(event.channel, event, event.member)
            event.jda.addEventListener(csm)
            Thread {
                try {
                    //timeout of 5 minutes
                    Thread.sleep(300000)
                    println(csm.done)
                    if (!csm.done) {
                        event.jda.removeEventListener(csm)
                    }
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }
            }.start()
        }
    }
}